okio/src/nonJvmMain/kotlin/okio/ByteString.kt

```diff
@@ -73,7 +88,10 @@ internal actual constructor(
   actual open fun substring(beginIndex: Int, endIndex: Int): ByteString =
     commonSubstring(beginIndex, endIndex)
 
-  internal actual open fun internalGet(pos: Int) = commonGetByte(pos)
+  internal actual open fun internalGet(pos: Int): Byte {
+    if (pos >= size || pos < 0) throw ArrayIndexOutOfBoundsException("size=$size pos=$pos")
```

Comment:
why is this changing? I think the whole motivation of _internal_ in the name is it’s not going to do a bounds check?

Reply:
This was a change made when consolidating all non jvm implementations of bytestring. This was directly copied from the (now deleted) js implementation of ByteString. I had to add this bounds check to throw the appropriate error. 

Without this, tests were failing. I can't remember the exact assertion error before, I'm going to replicate and see what it was.